core/src/main/scala/org/apache/spark/scheduler/local/LocalBackend.scala

```diff
@@ -74,10 +77,20 @@ private[spark] class LocalActor(
 
   def reviveOffers() {
     val offers = Seq(new WorkerOffer(localExecutorId, localExecutorHostname, freeCores))
-    for (task <- scheduler.resourceOffers(offers).flatten) {
-      freeCores -= scheduler.CPUS_PER_TASK
-      executor.launchTask(executorBackend, taskId = task.taskId, attemptNumber = task.attemptNumber,
-        task.name, task.serializedTask)
+    val tasks = scheduler.resourceOffers(offers).flatten
+    if (tasks.nonEmpty) {
```

Comment:
As long as you have to fix the comma issue, can you eliminate this "if" (just have the for-loop), and then on line 88, do "if (tasks.empty && scheduler.activeTaskSets.nonEmpty)"? I think that would make this slightly more resilient to being broken later on if someone doesn't notice the subtlety of the two if-statements here